components/locale_canonicalizer/src/locale_canonicalizer.rs

```diff
@@ -47,18 +120,276 @@ macro_rules! maximize_locale {
 
 impl LocaleCanonicalizer<'_> {
     /// A constructor which takes a [`DataProvider`] and creates a [`LocaleCanonicalizer`].
-    pub fn new<'d>(
-        provider: &(impl DataProvider<'d, LikelySubtagsV1> + ?Sized),
-    ) -> Result<LocaleCanonicalizer<'d>, DataError> {
-        let payload: DataPayload<LikelySubtagsV1> = provider
+    pub fn new<'d, P>(provider: &P) -> Result<LocaleCanonicalizer<'d>, DataError>
+    where
+        P: DataProvider<'d, crate::provider::AliasesV1>
+            + DataProvider<'d, crate::provider::LikelySubtagsV1>
+            + ?Sized,
+    {
+        let aliases: DataPayload<AliasesV1> = provider
+            .load_payload(&DataRequest::from(key::ALIASES_V1))?
+            .take_payload()?;
+
```

Comment:
Suggestion: Make a way to set up a LocaleCanonicalizer with the old LikelySubtags data without the new Aliases data, such as an option in an options bag.

Reply:
Agree. Since we seem to have a non-insignificant cost of adding aliases, and there may be users who don't care about aliases in their environment, having an ability to construct a canonicalizer that doesn't pay that cost seems worth it.

Reply:
That makes sense to me!